iothub/service/src/Utilities/Fx.cs

```diff
@@ -57,7 +57,7 @@ internal static bool IsFatal(Exception ex)
 
         internal static bool ContainsAuthenticationException(Exception ex)
         {
-            while (ex != null)
+            if (ex != null)
             {
                 return ex is AuthenticationException || ContainsAuthenticationException(ex.InnerException);
```

Comment:
The recursion is happening in here first, so it doesn't need the `while` loop at the top.